swan-lake/resources/featured-scenarios/build-a-data-service-in-ballerina.md

```diff
@@ -59,224 +55,229 @@ CREATE TABLE Company.Employees (
 
 ## Create the service package
 
-Ballerina uses packages to group code. You need to create a Ballerina package and write the business logic in it. In the terminal, execute the command below to create the Ballerina package for the implementation.
+Ballerina uses packages to group code. You need to create a Ballerina package and write the business logic in it. 
 
-> **Info:** For more information on Ballerina packages, see [Organize Ballerina code](/learn/organize-ballerina-code/).
+1. In the Terminal, execute the command below to create the Ballerina package for the implementation.
 
-```
-$ bal new data_service
-``` 
+    > **Info:** For more information on Ballerina packages, see [Organize Ballerina code](/learn/organize-ballerina-code/).
 
-This creates a directory named `data_service` with the files below.
+    ```
+    $ bal new build-a-data-service
+    ``` 
+    You should see the output similar to the following.
 
-```
-.
-├── data_service
-│   ├── Ballerina.toml
-│   └── main.bal
-```
+    ```
+    package name is derived as 'build_a_data_service'. Edit the Ballerina.toml to change it.
+
+    Created new package 'build_a_data_service' at /Users/build-a-data-service.
+    ```
+
+    This creates a directory named `build-a-data-service` with the files below.
 
->**Tip:** Remove the automatically-created `main.bal` file as you are not going to use it in this guide.
+    ```
+    .
+    │   ├── Ballerina.toml
+    │   └── main.bal
+    ```
 
-## Create the service
+2. Open the created package in VS Code.
+
+3. Remove the generated content in the `main.bal` file and open the diagram view in VS Code.
+
+    ![Open diagram view](/learn/images/featured-scenarios/build-a-data-service-in-ballerina/open-diagram-view.gif)
+
+## Create the data service
+
+The sections below demonstrate how to create the service.
 
 ### Create a record to represent an employee
 
-In Ballerina, records are a data type that maps keys to values. Define a closed record to represent a single row in the `Employees` table in the `main.bal` file. 
+In Ballerina, records are a data type that maps keys to values. Follow the steps below to define a closed record representing a single row in the `Employees` table in the `main.bal` file. 
 
 >**Info:** This record type is the basis for interacting with the database.
 
+Generate the record types corresponding to the payload from the service by providing the record name as `Employee` and the sample JSON object below.
+
+>**Tip:** You need to complete the generated record by adding the pipe signs to mark the record as a closed one, adding the `employee_id`, `manager_id`, and `hire_date` types, and importing the `ballerina/time` module, which cannot be represented in the JSON format.
+
+```json
+{    
+    "first_name": "test",
+    "last_name": "test",
+    "email": "test@test.com",
+    "phone": "882 771 110",
+    "job_title": "Sales Manager"
+}
```

Comment:
It is not good to have only half of payload and then ask to manually add the rest of the fields. Here in this approach users have to get modified payload as well , which is not the purpose of the feature. Users should be able to use the exact payload and then modify the field types and other optionality related things after that on the record. 

So the suggestion is copy the payalod as it is and then modify the record.

paylaod should be something like below  (Add with proper values): 

```json
{
  "employee_id": 0,
  "first_name": "string",
  "last_name": "string",
  "email": "string",
  "phone": "string",
  "hire_date": {
    "year": 0,
    "month": 0,
    "day": 0,
    "hour": 0,
    "minute": 0,
    "second": 0,
    "utcOffset": {
      "hours": 0,
      "minutes": 0,
      "seconds": 0
    }
  },
  "manager_id": 0,
  "job_title": "string"
}
```

Generated record will be 

```
type Employee record {
    int employee_id;
    string first_name;
    string last_name;
    string email;
    string phone;
    record {
        int year;
        int month;
        int day;
        int hour;
        int minute;
        int second;
        record {
            int hours;
            int minutes;
            int seconds;
        } utcOffset;
    } hire_date;
    int manager_id;
    string job_title;
};
```
Then it required to explain why replacing inner record with date time and why adding employee_id as optional filed and manager_id as optional type.

@MaryamZi wdyt?

Reply:
Yeah, +1. If the user is using the payload to generate the record they will use the entire payload. So at least `hire_date` and `manager_id`, which will always be present in the payload, should be included IMO.

OTOH, `employee_id`, which is optional, may or may not be in the sample payload they have. That we can ask to add separately if required.